### DISCUSSION TITLE
Rename file but not from scratch

### DISCUSSION THREAD

**Discussion by omnigenous** [gokcehan/lf#1762] [Q&A] [answered]
Sometime I get files with _really_ long names but I just need to change small part of it (ex: remove underscores) and retyping entire name prone to typos.

Here is my setup:

```bash
map r push :rename<space>
cmd rename %[ -e $1 ] && printf "file exists" || mv $f $1
```

When I press `r` `:rename ` appears at the bottom, can this be change to `:rename FILE_NAME.EXT` and/or `:rename FILE_NAME` (without ext).
I tried:

```diff
+ map r push :rename<space> $f
```

But that just adds literal `$f`.

**Comment by DusanLesan** [comment, marked as answer]
I don't know how to help with that, but I'm curious: what makes the built-in rename feature unsuitable for you?

**Reply by omnigenous** [reply]
Nothing, I don't remember why I implemented my own because I did go through entire wiki page in github but still did not know there built in rename.

https://github.com/gokcehan/lf/wiki/Tips#enhancing-the-rename-command
